### PR TITLE
Number of tasks completed per unit of time

### PR DESCRIPTION
**Pull request by oanaBiancaSchiopu**

### CODE REVIEW

File: scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/CompletedTasksCount.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+package org.ow2.proactive.scheduler.common.job;
+
+import java.io.Serializable;
+import java.util.Map;
+
+import org.objectweb.proactive.annotation.PublicAPI;
+
+
+@PublicAPI
+public class CompletedTasksCount implements Serializable {
```
[the remark this answers is not in the document]
Comment (oanaBiancaSchiopu) [reply scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/CompletedTasksCount.java:35]: an object of type CompletedTasksCount is returned when sending the request. If it does not implements Serializable I will get "java.io.NotSerializableException"

**Comment by mboussaa** [comment]
@oanaBiancaSchiopu LGTM but check with @fviale before merging because I don't understand much the goal of this PR

**Review by mboussaa — APPROVED** [review]

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java [hunk 2]
```diff
@@ -1278,6 +1315,37 @@ public CompletedJobsCount getCompletedJobs(String user, String tenant, final Str
...
+        long intervalTime = (endDate - startDate) / numberOfIntervals;
+
+        for (int interval = 0; interval < numberOfIntervals && startDate < endDate; interval++) {
+            long endTimeInterval = startDate + intervalTime;
+            Integer nrOfTasksWithoutIssues = getNumberOfFilteredTasks(taskName,
+                                                                      user,
+                                                                      tenant,
+                                                                      startDate,
+                                                                      endTimeInterval,
+                                                                      Collections.singletonList(TaskStatus.FINISHED));
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java:1332]: I think at least SKIPPED should be added in this list

Comment (fviale) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java:1332]: you can use ImmutableList.of(TaskStatus.FINISHED, TaskStatus.SKIPPED)

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java [hunk 3]
```diff
@@ -1278,6 +1315,37 @@ public CompletedJobsCount getCompletedJobs(String user, String tenant, final Str
...
+            tasksWithoutIssuesCount.put(interval, nrOfTasksWithoutIssues);
+            Integer nrOfFailedTasks = getNumberOfFilteredTasks(taskName,
+                                                               user,
+                                                               tenant,
+                                                               startDate,
+                                                               endTimeInterval,
+                                                               ImmutableSet.of(TaskStatus.ABORTED,
+                                                                               TaskStatus.FAILED,
+                                                                               TaskStatus.FAULTY,
+                                                                               TaskStatus.IN_ERROR));
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java:1342]: IN_ERROR is not a completed state, it's like PAUSED

Comment (fviale) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java:1342]: you should add NOT_STARTED and NOT_RESTARTED

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java [hunk 4]
```diff
@@ -391,9 +392,11 @@ public int getNumberOfFilteredJobs(final String workflowName, String user, Strin
             if (StringUtils.isNotEmpty(workflowName)) {
                 predicates.add(cb.like(root.get("jobName"), workflowName + "%"));
             }
-            if (startTime != 0 && endTime != 0) {
-                predicates.add(cb.and(cb.ge(root.get("finishedTime"), startTime),
-                                      cb.lt(root.get("finishedTime"), endTime)));
+            if (startTime != 0) {
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java:395]: shouldn't this test be > 0 as -1 is surely not a good value

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java [hunk 5]
```diff
@@ -391,9 +392,11 @@ public int getNumberOfFilteredJobs(final String workflowName, String user, Strin
...
+                predicates.add(cb.ge(root.get("finishedTime"), startTime));
+            }
+            if (endTime != 0) {
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java:398]: same comment as above

**Pull request merged**